ethstaker_deposit/key_handling/key_derivation/mnemonic.py

```diff
@@ -123,17 +136,17 @@ def reconstruct_mnemonic(mnemonic: str, words_path: str) -> Optional[str]:
             entropy_bits = entropy.to_bytes(checksum_length * 4, 'big')
             full_word_list = _get_word_list(language, words_path)
             if _get_checksum(entropy_bits) == checksum:
-                """
-                This check guarantees that only one language has a valid mnemonic.
-                It is needed to ensure abbrivated words aren't valid in multiple languages
-                """
-                if reconstructed_mnemonic is not None:
-                    raise ValidationError("This mnemonic abbreviated form is available in multiple languages.")
+                valid_languages.append(language)
                 reconstructed_mnemonic = ' '.join([_index_to_word(full_word_list, index) for index in word_indices])
             else:
                 pass
         except ValueError:
             pass
+
+    if len(valid_languages) > 1:
+        valid_languages.sort()
```

Comment:
Sorting here should be based not on the languages string value but on the canonical sort order we are using elsewhere which is defined in `ethstaker_deposit/utils/constants.py:MNEMONIC_LANG_OPTIONS`

Reply:
ack